gen/cycle.ml

```diff
@@ -24,27 +24,37 @@ module type S = sig
   type atom
   module PteVal : PteVal_gen.S with type pte_atom = atom
 
+  (* TODO can be parametric by dir *)
   type event =
-      { loc : loc ; ord : int; tag : int ;
+      { loc : loc ; ord : int; 
+        (* TODO morello related value, fold into value *)
+        tag : int ;
         ctag : int; cseal : int; dep : int ;
         v   : v ; (* Value read or written *)
+        (* TODO fold into value *)
         vecreg: v list list ; (* Alternative for SIMD *)
+        (* TODO instruction fetch related value, fold into value *)
         ins : int ;
         dir : dir option ;
         proc : Code.proc ;
         atom : atom option ;
+        (* TODO why not put into dir ? *)
         rmw : bool ;
+        (* TODO, cell and tcell fold together *)
         cell : v array ; (* Content of memory, after event *)
         tcell : v array ; (* value of tag memory after event *)
         bank : SIMD.atom Code.bank ;
         idx : int ;
+        (* TODO PTE fold into value *)
```

Comment:
is value referring to the v in this structure? it is a little unclear to me from the comment alone.

Reply:
Yes it means to fold into value type `v`. I will make it clear. I am working on it now as a side code improvement project but will make a separate pull once it is ready.